src/components/Contact/contactform.tsx

```diff
@@ -0,0 +1,280 @@
+import React, { useEffect } from 'react'
+import { Dispatch, SetStateAction, useState } from 'react'
+import Image from 'next/image'
+import { PlusIcon, UserCircleIcon, XIcon } from '@heroicons/react/outline'
+import tw from 'tailwind-styled-components'
+import { v4 as uuidv4 } from 'uuid'
+
+import type { Contact, Pet } from '@/types/types'
+type ContactInfoProps = {
+  contact: Contact
+  image: string
+  setIsEditing: Dispatch<SetStateAction<boolean>>
+}
+import RegionSelector from '@/components/Contact/regiondropdown'
+import TagSelector from '@/components/Contact/tagdropdown'
+
+const ContactForm = ({ contact, image, setIsEditing }: ContactInfoProps) => {
+  const [pets, setPets] = useState<Pet[]>(contact.pets)
+  const [regions, setRegions] = useState(contact.region)
+  const [tags, setTags] = useState(contact.tags)
+  const [contactForm, setContactForm] = useState(contact)
+
+  function addPet() {
+    const newPetField = {
+      id: uuidv4(),
+      name: '',
+      notes: ''
+    }
+    setPets((pets) => [...pets, newPetField])
+  }
+
+  const removePet = (pet: Pet) => {
+    setPets([...pets.filter((p) => p.name !== pet.name)])
+  }
+
+  const handleInputChange = (
+    e:
+      | React.ChangeEvent<HTMLInputElement>
+      | React.ChangeEvent<HTMLTextAreaElement>
+  ) => {
+    const { name, value } = e.target
+    setContactForm({ ...contactForm, [name]: value })
+  }
+
+  const handlePetChange = (
+    petIndex: number,
+    e:
+      | React.ChangeEvent<HTMLInputElement>
+      | React.ChangeEvent<HTMLTextAreaElement>
+  ) => {
+    const { name, value } = e.target
+    const newPets = pets.map((pet, i) =>
```

Comment:
If only one element with a given index needs to be modified, map() won't be needed